opennlp-tools/src/main/java/opennlp/tools/postag/POSTaggerFactory.java

```diff
@@ -363,11 +382,11 @@ public void validateArtifactMap() throws InvalidFormatException {
     if (ngramDictEntry != null && !(ngramDictEntry instanceof Dictionary)) {
       throw new InvalidFormatException("NGram dictionary has wrong type!");
     }
-
   }
 
+  // reduced visibility to ensure deprecation is respected in future versions
   @Deprecated
```

Comment:
We are more strict about changing visibility or removing public methods in Commons between non-major releases, but I think it should be fine for OpenNLP, as its API is not expected to be used by most users (i.e. most should be using the tools to train/infer with models instead). Even then, worth adding a note to the changelog about it IMO.

Reply:
I've been thinking around at this line for a while. There are so many places that are deprecated _and_ unused. Worth documenting things that are both? Or remove or change the visibility so future releases mildly switch users to change their code to drop deprecated stuff that has been in this state for many years.

Agreed that we need a policy on how to clean up or handle (internally unused!) Deprecated things.

Same applies for all other spots I felt an adjustment makes sense.

Reply:
Yeah, I know Commons really tries to be as backward compatible as possible, even avoiding updating the minimum JVM version. Jena also cares about breaking certain parts of the public API that are known/expected to be used in other systems.

I don't know about OpenNLP though. +1 to having it well documented so devs/committers are aware of what can be done or not. Thanks for understanding it @mawiesne !

Reply:
Sure. IMHO, it's important to define under which circumstances deprecation is tolerable, or in other words how to evolve an API. I've seen spots in the OpenNLP code where only a JavaDoc comment signals deprecation, so the compiler wouldn't warn devs (external or "homies").

Reply:
With @jzonthemtn in mind, I'm open for both directions for this PR here. Maybe, Jeff can quickly check if in this case the change in `POSModelFactory` can be done without hurting OpenNLP community (too) much. If it hurts: I'll try to go back and leave that "init..." part untouched.

Reply:
> +1 to having it well documented so devs/committers are aware of what can be done or not.

+1 -  we need some clear guideance / documentation. People need to know what to expect if they are using deprecated things. Judging from the cleanup process, we deprecated things (e.g. via a Javadoc comment) but never get down the road to actually cleanup or remove them ;-)

> Sure. IMHO, it's important to define under which circumstances deprecation is tolerable, or in other words how to evolve an API. I've seen spots in the OpenNLP code where only a JavaDoc comment signals deprecation, so the compiler wouldn't warn devs (external or "homies").

I am +1 with @mawiesne  - the best thing would be, that we find some consent on how to deal with deprecated things between releases and document it, so we have some written consent, which isn't forgotten or only in the "heads" :-)

That said, I really appreciate @mawiesne effort to increase the readability & clarity of the code! :)